cpp/example_code/cross-service/topics_and_queues/README.md

```diff
@@ -0,0 +1,170 @@
+
+Use the sample code in this folder to explore publishing and subscribing to a topic by using filters and queues. This tutorial does not create a complete end-to-end application. Instead, you can use it to play around with a publish and subscribe architecture.
+
+You can create an Amazon SNS topic and subscribe two Amazon SQS queues to the topic. You can enable FIFO (First-In-First-Out) queueing, and you can add filtered subscriptions. Then, you can publish messages to the topic and see the results in the queues.
+
+You can publish and subscribe using just SNS. But combining SNS with SQS gives you more flexibility in how the messages are consumed.
+
+SNS is a push service. It pushes to endpoints such as email addresses, mobile application endpoints, or SQS queues. (For a full list of endpoints, see [SNS event destinations](https://docs.aws.amazon.com/sns/latest/dg/sns-event-destinations.html)).
+
+With SQS, messages are received from a queue by polling. With polling, the subscriber receives messages by callling a receive message api. Any code can poll the queue. Also, the messages stay in the queue until you delete them. This gives you more flexibility in how the messages are processed.
+
+The sample code builds a command line application that asks you for input. This is implemented in multiple programming languages, and the interface can vary slightly between languages. The following shows the interface for the C++ implementation.
+
+### Create an SNS topic
+
+```
+Would you like to work with FIFO topics? (y/n) 
+```
+
+You configure FIFO (First-In-First-Out) topics when you create them. Choosing a FIFO topic enables other options, too. To learn more, see [FIFO topics example use case](https://docs.aws.amazon.com/sns/latest/dg/fifo-example-use-case.html).
+
+
+```
+Would you like to use content-based deduplication instead of a deduplication ID? (y/n)
+```
+
+Deduplication is only available for FIFO topics. Deduplication prevents the subscriber from responding more than once to events that are determined to be duplicates. If a message gets published to an SNS FIFO topic and it’s found to be a duplicate within the five-minute deduplication interval, the message is accepted but not delivered. For more information, see [Message deduplication for FIFO topics](https://docs.aws.amazon.com/sns/latest/dg/fifo-message-dedup.html).
+
+Content-based deduplication uses a hash of the content as a deduplication ID. If content-based deduplication is not enabled, you must include a deduplication ID with each message.
+
+```
+Enter a name for your SNS topic:
+```
+
+Topic names can have 1-256 characters. They can contain uppercase and lowercase ASCII letters, numbers, underscores, and hyphens. If you chose a FIFO topic, the application automatically adds a “.fifo” suffix, which is required for FIFO topics.
+
+### Create two SQS queues.
+
+Now, configure two SQS queues to subscribe to your topic.
```

Comment:
Optional, but it's a little unclear at this point why there are 2 queues. Should we describe why there are 2?

Reply:
Good idea. I added a description.

Reply:
Now, configure two SQS queues to subscribe to your topic. Using a separate queue for each subscriber can be useful. You can customize how messages are consumed and
how messages are filtered, for instance.